tests/test_utility.py

```diff
@@ -20,66 +20,115 @@
+        ansConfigDir = MTAOS.getModulePath().joinpath("policy")
+        self.assertEqual(MTAOS.getConfigDir(), ansConfigDir)
 
     def testGetSchemaDir(self):
 
-        ansSchemaDir = getModulePath().joinpath("schema")
-        self.assertEqual(getSchemaDir(), ansSchemaDir)
+        ansSchemaDir = MTAOS.getModulePath().joinpath("schema")
+        self.assertEqual(MTAOS.getSchemaDir(), ansSchemaDir)
+
+    def testGetLogDir(self):
+
+        ansLogDir = MTAOS.getModulePath().joinpath("logs")
+        self.assertEqual(MTAOS.getLogDir(), ansLogDir)
 
     def testGetIsrDirPathNotAssigned(self):
 
-        isrDir = getIsrDirPath()
+        isrDir = MTAOS.getIsrDirPath()
         self.assertEqual(isrDir, None)
 
     def testGetIsrDirPath(self):
 
         ISRDIRPATH = "/path/to/isr/dir"
         os.environ["ISRDIRPATH"] = ISRDIRPATH
 
-        isrDir = getIsrDirPath()
+        isrDir = MTAOS.getIsrDirPath()
         self.assertEqual(isrDir, Path(ISRDIRPATH))
 
         os.environ.pop("ISRDIRPATH")
 
     def testGetCamType(self):
 
-        self.assertEqual(getCamType("lsstCam"), CamType.LsstCam)
-        self.assertEqual(getCamType("lsstFamCam"), CamType.LsstFamCam)
-        self.assertEqual(getCamType("comcam"), CamType.ComCam)
+        self.assertEqual(MTAOS.getCamType("lsstCam"), CamType.LsstCam)
+        self.assertEqual(MTAOS.getCamType("lsstFamCam"), CamType.LsstFamCam)
+        self.assertEqual(MTAOS.getCamType("comcam"), CamType.ComCam)
 
-        self.assertRaises(ValueError, getCamType, "wrongType")
+        self.assertRaises(ValueError, MTAOS.getCamType, "wrongType")
 
     def testGetInst(self):
 
-        self.assertEqual(getInstName("lsst"), InstName.LSST)
-        self.assertEqual(getInstName("comcam"), InstName.COMCAM)
-        self.assertEqual(getInstName("sh"), InstName.SH)
-        self.assertEqual(getInstName("cmos"), InstName.CMOS)
+        self.assertEqual(MTAOS.getInstName("lsst"), InstName.LSST)
+        self.assertEqual(MTAOS.getInstName("comcam"), InstName.COMCAM)
+        self.assertEqual(MTAOS.getInstName("sh"), InstName.SH)
+        self.assertEqual(MTAOS.getInstName("cmos"), InstName.CMOS)
 
-        self.assertRaises(ValueError, getInstName, "wrongName")
+        self.assertRaises(ValueError, MTAOS.getInstName, "wrongName")
 
     def testGetCscName(self):
 
-        cscName = getCscName()
+        cscName = MTAOS.getCscName()
         self.assertEqual(cscName, "MTAOS")
 
+    def testAddRotFileHandler(self):
+
+        log = logging.Logger("test")
+        dataDirPath = self.dataDir.name
+        filePath = Path(dataDirPath).joinpath("test.log")
+        MTAOS.addRotFileHandler(log, filePath, maxBytes=1e3, backupCount=5)
+
+        handlers = log.handlers
+        self.assertEqual(len(handlers), 1)
+        self.assertTrue(isinstance(handlers[0], RotatingFileHandler))
+
+        for counter in range(20):
+            log.critical("Test file rotation.")
+            time.sleep(0.2)
```

Comment:
Is such a long delay necessary? If 0.01 would do then you could speed up this test.

Reply:
I tested the different delay times (1 sec to 0.01 sec) on my macbook Pro, Linux laptop, local Linux desktop, and Jenkins already. I am shocked to see the `log` can not write to the log files for all messages (I tried to use the `range(100)` before). Therefore, I set the time to be `0.2` sec to be safe to write all the test data and I do not need to wait long.
I am guessing when `log` write the data into the file, it always open/ close the file for each new line. Otherwise, I can not explain what I saw.
Let me keep this sleep time at this moment since it does not affect the result.

Reply:
Of course. I'm glad you tested.